### DISCUSSION TITLE
Use `cf-connecting-ip` as `proxy.clientIp` in log drain?

### DISCUSSION THREAD

**Discussion by b0o** [vercel/community#2572] [Help] [answered]
Because our Next.js app on Vercel is proxied through Cloudflare, our access logs show Cloudflare IPs instead of actual client IPs. Is it possible to configure the data sent to a log drain to include the value of a header like `cf-connecting-ip` or `x-real-ip`?

**Comment by smaeda-ks** [comment, marked as answer]
Hi @b0o, that's an enterprise-only feature as of today. Instead, you can send a custom request header via Cloudflare containing the actual client IP address and log that value in your middleware/function.

**Reply by b0o** [reply]
Thank you for the response. I hope that this feature can be rolled out to Pro and/or Hobby plans in the future.
